airbyte-config/models/src/main/java/io/airbyte/config/helpers/RawListToStandardXDefinitions.java

```diff
@@ -0,0 +1,130 @@
+      new SimpleImmutableEntry<>(StandardSourceDefinition.class.getCanonicalName(), "sourceDefinitionId"));
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  public static List<StandardSourceDefinition> toStandardSourceDefinitions(String yamlStr) throws RuntimeException {
+    return verifyAndConvertToModelList(StandardSourceDefinition.class, yamlStr);
+  }
+
+  public static List<StandardDestinationDefinition> toStandardDestinationDefinitions(String yamlStr) throws RuntimeException {
+    return verifyAndConvertToModelList(StandardDestinationDefinition.class, yamlStr);
+  }
+
+  public static JsonNode verifyAndConvertToJsonNode(String idName, String yamlStr) throws RuntimeException {
+    final var jsonNode = Yamls.deserialize(yamlStr);
+    checkYamlIsPresentWithNoDuplicates(jsonNode, idName);
+    return jsonNode;
+  }
+
+  @VisibleForTesting
+  static <T> List<T> verifyAndConvertToModelList(Class<T> klass, String yamlStr) throws RuntimeException {
+    final var jsonNode = Yamls.deserialize(yamlStr);
+    final var idName = classNameToIdName.get(klass.getCanonicalName());
+    checkYamlIsPresentWithNoDuplicates(jsonNode, idName);
+    return toStandardXDefinitions(jsonNode.elements(), klass);
+  }
+
+  private static void checkYamlIsPresentWithNoDuplicates(JsonNode deserialize, String idName) throws RuntimeException {
+    final var presentDestList = !deserialize.elements().equals(ClassUtil.emptyIterator());
+    Preconditions.checkState(presentDestList, "Definition list is empty");
+    checkNoDuplicateNames(deserialize.elements());
+    checkNoDuplicateIds(deserialize.elements(), idName);
+  }
+
+  private static void checkNoDuplicateNames(final Iterator<JsonNode> iter) throws IllegalArgumentException {
+    final var names = new HashSet<String>();
+    while (iter.hasNext()) {
+      final var element = Jsons.clone(iter.next());
+      final var name = element.get("name").asText();
+      if (names.contains(name)) {
+        throw new IllegalArgumentException("Multiple records have the name: " + name);
+      }
+      names.add(name);
+    }
+  }
+
+  private static void checkNoDuplicateIds(final Iterator<JsonNode> fileIterator, final String idName) throws IllegalArgumentException {
+    final var ids = new HashSet<String>();
+    while (fileIterator.hasNext()) {
+      final var element = Jsons.clone(fileIterator.next());
+      final var id = element.get(idName).asText();
+      if (ids.contains(id)) {
+        throw new IllegalArgumentException("Multiple records have the id: " + id);
+      }
+      ids.add(id);
+    }
+  }
+
+  private static <T> List<T> toStandardXDefinitions(Iterator<JsonNode> iter, Class<T> c) throws RuntimeException {
+    Iterable<JsonNode> iterable = () -> iter;
+    var defList = new ArrayList<T>();
+    for (JsonNode n : iterable) {
+      try {
+        var def = mapper.treeToValue(n, c);
```

Comment:
I think you can use `Jsons.object(...)` here.

Reply:
Nice catch!